src/Native/Runtime/arm/PInvoke.S

```diff
@@ -30,46 +18,53 @@
 // Also, the codegenerator must ensure that there are no live GC references in callee saved registers.
 //
 NESTED_ENTRY RhpPInvoke, _TEXT, NoHandler
-	        PROLOG_PUSH "{r4-r7,lr}"
+        str         lr, [r0, #OFFSETOF__PInvokeTransitionFrame__m_RIP]
+        str         r7, [r0, #OFFSETOF__PInvokeTransitionFrame__m_FramePointer]
+        str         sp, [r0, #OFFSETOF__PInvokeTransitionFrame__m_PreservedRegs]
+
+        PROLOG_PUSH "{r5-r7,lr}"
 
-	        str         lr, [r0, #OFFSETOF__PInvokeTransitionFrame__m_RIP]
-	        str         r7, [r0, #OFFSETOF__PInvokeTransitionFrame__m_FramePointer]
-	        str         sp, [r0, #OFFSETOF__PInvokeTransitionFrame__m_PreservedRegs]
+        // get TLS global variable address
+        mov         r5, r0
+        ldr         r0, .TLS_CURRENT_THREAD_LABEL
```

Comment:
It would be nice to keep this in the macro (and the helper var below as well) - the access to this thread local var is repeated in many asm helpers.